.github/workflows/ci-nf-test.yml

```diff
@@ -0,0 +1,49 @@
+name: nf-test CI
+# Run nf-test
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+
+
+jobs:
+  test:
+    name: nf-test, NF ${{ matrix.NXF_VER }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        NXF_VER:
+          - "23.10.1"  
```

Comment:
Since the tests tend to be slow, I only run them with the expected nextflow version in production. 
However, it is possible to add more nextflow versions if necessary.